content/utilities/mark-friends/plugin.js

```diff
@@ -0,0 +1,279 @@
+    borderRadius: "3px",
+  };
+
+  let planetLevelStyle = {
+    marginTop: "10px",
+    marginBottom: "10px"
+  };
+
+  const minLevelSelect = html`
+  <select
+	style=${selectStyle}
+	value=${inMinLevel}
+	onChange=${(e) => {
+      minLevel = Number(e.target.value);
+      setInMinLevel(Number(e.target.value))
+    }
+    }
+  >
+	${PLANET_LEVELS.map(
+      ({ value, text }) => html`<option value=${value}>${text}</option>`
+    )}
+  </select>
+`;
+
+  const maxLevelSelect = html`
+<select
+  style=${selectStyle}
+  value=${inMaxLevel}
+  onChange=${(e) => {
+      maxLevel = Number(e.target.value);
+      setInMaxLevel(Number(e.target.value))
+
+    }
+    }
+>
+  ${PLANET_LEVELS.map(
+      ({ value, text }) => html`<option value=${value}>${text}</option>`
+    )}
+</select>
+`;
+
+  let friendChildren =
+    inPlanet.map((item) => {
+      return html`<div>
+		${shortAddress(item.owner)}
+		${' '}
+		<button style=${buttonStyle}
+		 onClick=${() => { center(item) }}
+		> center </button>
+		${' '}
+		<button
+		style=${buttonStyle}
+		onClick=${() => { removeFriend(item); }}
+		> delete </button>
+	  </div>`;
+    });
+
+  function aimPlanet() {
+    let tmpPlanet = ui.getSelectedPlanet();
+    setChosenPlanet(tmpPlanet);
+  }
+
+  function center(thePlanet) {
+    ui.centerLocationId(thePlanet.locationId);
+  }
+
+  function addFriend() {
+    if (chosenPlanet == undefined) return;
+    let flag = true;
+    for (let i in planet) {
+      if (planet[i].owner == chosenPlanet.owner)
+        flag = false;
+    }
+
+    if (flag == true) planet.push(chosenPlanet);
+    setInPlanet([...planet]);
+    friend = [];
+    for (const i in planet)
+      friend.push(planet[i].owner);
+  }
+
+
+  function removeFriend(thePlanet) {
+    let newPlanet = [];
+    for (let i in planet) {
+      let p = planet[i];
+      if (p.owner == thePlanet.owner) continue;
+      newPlanet.push(p);
+    }
+    planet = newPlanet;
+    setInPlanet([...planet]);
+    friend = [];
+    for (const i in planet) {
+      friend.push(planet[i].owner);
+    }
+  }
+
+  function clearFriend() {
+    planet = [];
+    friend = [];
+    setInPlanet(planet);
+  }
+
+  return html`<div  style=${divStyle} >
+
+	  <button onClick=${aimPlanet}> choose planet </button>
+	  <h1> ${chosenPlanet == undefined ? "not choose yet" : chosenPlanet.owner}</h1>
+
+	  <div> 
+
+	  <button onClick=${addFriend}>add friend</button>
+	  ${' '}
+	  <button onClick=${clearFriend}>clear all</button>
+	  </div>
+	  <div>${' '}</div>
+
+	  <div 
+	  style=${planetLevelStyle}
+
+		> 
+		<h1> choose the planet Level </h1>
+
+		${minLevelSelect}
+		${' '}
+		${maxLevelSelect}
+
+	  </div>
+
+	  <div> number of friends： ${inPlanet.length}</div>
+	  <div style=${listStyle}
+	  > ${friendChildren}</div>
+  </div>`;
+}
+
+
+function App() {
+  return html`<${friendList} />`;
+}
+
+class Plugin {
+  constructor() {
+    if (typeof window.planet === "undefined") {
+      window.planet = [];
```

Comment:
This should be namespaced behind your plugin name or something, like window.MARK_FRIENDS.planet